packages/discord.js/typings/index.test-d.ts

```diff
@@ -1358,7 +1358,11 @@ new SelectMenuComponent({
 
 new ButtonComponent({
   style: ButtonStyle.Danger,
-});
+})
+  .setEmoji('<a:foo:123>')
+  .setEmoji('<:foo:123>')
+  // @ts-expect-error
+  .setEmoji(':foobar:123');
```

Comment:
Util.parseEmoji can also match `foobar:123`, `:foobar:123`, etc. Might want to update types if we want to be extra strict, or just keep it as string

Reply:
It should allow default emojis too. So imo string will be better option.

Reply:
It should just allow everything that parseEmoji allows, just like in v13

Reply:
> It should just allow everything that parseEmoji allows, just like in v13
It should also allow `APIMessageComponentEmoji`.